lib/Conversion/ImportVerilog/Expressions.cpp

```diff
@@ -1124,3 +1114,32 @@ Value Context::materializeConversion(Type type, Value value, bool isSigned,
     value = builder.create<moore::ConversionOp>(loc, type, value);
   return value;
 }
+
+void Context::traverseUnpacked(
+    const slang::ast::FixedSizeUnpackedArrayType &slangType, Value value,
+    SmallVector<Value> &conditions, Value lhs, Location loc) {
+  Value cond;
+  // auto &slangType =
+  // listExpr->type->as<slang::ast::FixedSizeUnpackedArrayType>();
+  auto mooreType = dyn_cast<moore::UnpackedArrayType>(convertType(slangType));
+  const auto &elementType = slangType.elementType;
+  for (slang::int32_t i = slangType.getFixedRange().lower();
+       i <= slangType.getFixedRange().upper(); i++) {
+    auto elemValue = builder.create<moore::ExtractOp>(
+        loc, mooreType.getElementType(), value, i);
+    if (elementType.isUnpackedArray()) {
+      traverseUnpacked(elementType.as<slang::ast::FixedSizeUnpackedArrayType>(),
+                       elemValue, conditions, lhs, loc);
+    } else if (elementType.isSingular()) {
+      if (elementType.isIntegral()) {
+        cond = builder.create<moore::WildcardEqOp>(loc, lhs, elemValue);
+      } else {
+        cond = builder.create<moore::EqOp>(loc, lhs, elemValue);
+      }
+      conditions.push_back(cond);
+    } else {
+      mlir::emitError(loc, "only singular or sized unpacked arrays supported "
+                           "in unpacked arrays in 'inside' expressions");
```

Comment:
Коряво получилось. Предлагаю переписать иначе, например вот так:

```suggestion
      mlir::emitError(loc, "only singular values and fixed-size unpacked arrays "
                           "allowed as elements of unpacked arrays in 'inside' "
                           "expressions");
```